Objects/bytearrayobject.c

```diff
@@ -707,6 +708,11 @@ static int
 bytearray_ass_subscript_lock_held(PyObject *op, PyObject *index, PyObject *values)
 {
     _Py_CRITICAL_SECTION_ASSERT_OBJECT_LOCKED(op);
+#ifdef Py_DEBUG
+    if (values != NULL) {
+        _Py_CRITICAL_SECTION_ASSERT_OBJECT_LOCKED(values);
+    }
+#endif
```

Comment:
I don't think this needs the `Py_DEBUG` guard, `_Py_CRITICAL_SECTION_ASSERT_OBJECT_LOCKED` is already a noop on non-debug builds and I'd assume the compiler will optimize away the empty branch.

Reply:
Agreed but was told to do it this way by other contributor, will remove.